api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -39,6 +39,13 @@
    */
   int specId();
 
+  /**
+   * Return id of the schema when write this file.
+   */
+  default int schemaId() {
+    return -1;
```

Comment:
Instead of returning an incorrect value, I think this should throw an exception.

Reply:
I tried with throw an exception and then we need to add more `try catch` the wrap the function call. I keep this with -1 and add comments for the -1 means.

Reply:
None of the implementations should actually use this implementation. The `default` is needed for binary compatibility, but if anyone calls this it should throw an exception.

Reply:
Updated